switchsdk/src/androidTest/java/net/gini/switchsdk/uitests/TakePictureScreenTest.java

```diff
@@ -0,0 +1,236 @@
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class TakePictureScreenTest {
+
+    private static final int TIME_TO_WAIT_BETWEEN_STEPS = 1000;
+    private static final int TIME_TO_WAIT_FOR_ACTIVITY_TO_START = 4000;
+    @Rule
+    public IntentsTestRule<HostActivity> mActivityTestRule = new IntentsTestRule
+            <>(HostActivity.class, true, false);
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(
+            Manifest.permission.CAMERA);
+    private Context mTargetContext;
+
+    @Test
+    public void cameraScreen_rejectedPictureShouldNotAppearInDocumentBar() {
+
+        omitOnboarding();
+        startIntent();
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+        onView(withId(R.id.button_take_picture))
+                .perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_FOR_ACTIVITY_TO_START);
+
+        onView(withId(R.id.button_discard)).perform(click());
+
+        SystemClock.sleep(1000);
+
+        onView(withId(R.id.image_overview)).check(new RecyclerViewItemCountAssertion(1));
+
+    }
+
+    @Test
+    public void cameraScreen_takePictureShouldOpenReviewScreen() {
+
+        omitOnboarding();
+        startIntent();
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+        onView(withId(R.id.button_take_picture))
+                .perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_FOR_ACTIVITY_TO_START);
+        intended(hasComponent(ReviewPictureActivity.class.getName()));
+    }
+
+    @Test
+    public void firstLaunch_shouldShowOnboarding() {
+
+        startIntent();
+
+        SystemClock.sleep(3000); //TODO verify that animations on test device has been disabled
+        onView(withId(R.id.onBoardingContainer)).perform(swipeLeft()).check(matches(isDisplayed()));
+        onView(withId(R.id.onBoardingContainer)).perform(swipeLeft()).check(matches(isDisplayed()));
+        onView(withId(R.id.onBoardingContainer)).perform(swipeLeft()).check(
+                matches(withEffectiveVisibility(ViewMatchers.Visibility.GONE)));
+
+    }
+
+    @Test
+    public void menu_cancelShouldDisplayDialog() {
+
+        omitOnboarding();
+
+        startIntent();
+
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+
+        ViewInteraction appCompatTextView = onView(
+                allOf(withId(R.id.title), withText(R.string.menu_entry_cancel), isDisplayed()));
+        appCompatTextView.perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+
+        onView(withText(R.string.exit_dialog_text))
+                .inRoot(isDialog())
+                .check(matches(isDisplayed()));
+
+        onView(withId(android.R.id.button1))
+                .inRoot(isDialog())
+                .check(matches(withText(R.string.confirm)))
+                .check(matches(isDisplayed()));
+
+        onView(withId(android.R.id.button2))
```

Comment:
You already have those condition checks for the overflow menu test in `ReviewPictureScreenTest.java`. When I was making the UI tests, it was really helpful to extract such actions (opening a menu, verifying a component is there, going to a screen etc.) in a function. They are not very independent because they assume the app is in a certain state, but once you have them, you can chain them together in a test, avoid duplication and make the test really readable.